cairo/ethereum/cancun/vm/instructions/system.cairo

```diff
@@ -437,7 +442,10 @@ func generic_create{
     }
 
     let account_has_code_or_nonce_ = account_has_code_or_nonce{state=state}(contract_address);
-    if (account_has_code_or_nonce_.value != 0) {
+    let account_has_storage_ = account_has_storage{poseidon_ptr=poseidon_ptr, state=state}(
```

Comment:
how come EELs / here we don't use `is_account_empty`

Reply:
```python
if account_has_code_or_nonce(
        evm.env.state, contract_address
    ) or account_has_storage(evm.env.state, contract_address):
        increment_nonce(evm.env.state, evm.message.current_target)
        push(evm.stack, U256(0))
        return
```
Here is the condition for EELS.

is_account_empty is only for checking if an account has zero nonce, empty code and zero balance which is different from having storage

Reply:
ah sorry missread, indeed